### PR TITLE
Add github processes doc to IF repo

### PR DESCRIPTION
**Pull request by jmcook1186**
### Types of changes

Adds a markdown doc to the top level directory detailing our internal Github processes

### CODE REVIEW

File: github-processes.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+- npm packages are released using `release` branch
+- PRs into `release` are forbidden except to update tests, README, Github config, CI/CD or release config.
+- Pushing directly to `release` is forbidden
+- `release` should always satisfy a basic set of regression and scenario tests
+- merging `main` into `release` requires that all automated tests pass and two core team members have approved, one being QA.
+
+### `if-plugin-template` 
+
+- we only maintain a single branch: `main`
+- PRs to `main` can be mrged after one approval from core team
```
Comment (manushak) [inline github-processes.md:53]: typo in 'mrged'

File: github-processes.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+- [`if`](https://github.com/Green-Software-Foundation/if)
+    - source code for the IF
+- [`if-plugins`](https://github.com/Green-Software-Foundation/if-plugins)
+    - source code for standard library of plugins
+    - IF core team commit to maintaining these plugins 
+- [`if-unofficial-plugins`](https://github.com/Green-Software-Foundation/if-unofficial-plugins)
+    - source code for plugins relying on third party data/APIs
+    - intended to be deprecated and removed in mid-term future
+    - plugins int his repo should be handed over to relevant orgs to maintain
```
Comment (manushak) [inline github-processes.md:12]: 'int his' should be 'in this'

File: github-processes.md (hunk 1 above)
Comment (jmcook1186) [reply github-processes.md:53]:
```suggestion
- PRs to `main` can be merged after one approval from core team
```

File: github-processes.md (hunk 2 above)
Comment (jmcook1186) [reply github-processes.md:12]:
```suggestion
    - plugins in this repo should be handed over to relevant orgs to maintain
```

File: github-processes.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+// priority labels 
+high-priority 
+med-priority 
+low-priority 
+
+//delay explanation labels 
+abandoned 
+blocked 
+backlogged
```
Comment (jmcook1186) [inline github-processes.md:133]:
```suggestion
backlogged

// other labels
project-management
agenda
discussion
help-wanted
good-first-issue
```

File: github-processes.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+// other labels
+project-management
+agenda
+discussion
+help-wanted
+good-first-issue
+```
+
+Each PR is expected to have one label from each category assigned to it at all times.
```
Comment (jmcook1186) [inline github-processes.md:143]:
```suggestion
Each PR is expected to have one label from each category (except `other`) assigned to it at all times.
```

**Review by manushak — APPROVED** [review]

**Review by narekhovhannisyan — APPROVED** [review]

**Pull request merged**